gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/GemFireCacheImpl.java

```diff
@@ -1066,7 +1066,19 @@ public void run() {
       }
       this.memorySize = memorySize;
       if (memorySize > 0) {
-        this.bufferAllocator = ManagedDirectBufferAllocator.instance().initialize();
+        try {
+          Class clazz = Class.forName("com.gemstone.gemfire.internal.cache.store.ManagedDirectBufferAllocator");
+          Method method = clazz.getDeclaredMethod("instance");
+          this.bufferAllocator = (DirectBufferAllocator)method.invoke(null);
+        } catch (ClassNotFoundException e) {
+          throw new IllegalArgumentException("The property memory-size is not supported in SnappyData OSS version.");
+        } catch (NoSuchMethodException e) {
+          throw new IllegalStateException("Could not configure buffer allocator.", e);
```

Comment:
Any reason for catching NoSuchMethod, IllegalAccess and InvocationTarget and wrapping them in a IllegalState? These errors are self explanatory.